jenkins/Jenkinsfile

```diff
@@ -1,15 +1,25 @@
-// Update the custom data proc image used in leo.  The name of the job contains the name of the dataproc image created. This job will fail if master of terra-docker has not been updated since the last run, as the git hash has not changed. This is a happy side-effect of the fact image names must be unique.
```

Comment:
Is this `Jenkinsfile` intended to support GCE images _and_ snapshots or just snapshots? 

If it just supports snapshots then maybe it should be a separate `Jenkinsfile` so it does not break the existing https://fc-jenkins.dsp-techops.broadinstitute.org/job/leonardo-custom-image/ job (which reads the `Jenkinsfile` from `develop` branch I think).

Reply:
This `Jenkinsfile` is intended to eventually support both GCE snapshot and custom Dataproc image generation. Since we'll be able to cut over to snapshots for GCE pretty soon, I was hoping to not have two `Jenkinsfile`s. We could still run the previously existing Jenkins job (`leonardo-custom-image`) to generate any GCE custom images in case something comes up in the meantime. Once we cut over, we can just remove that Jenkins job.

Reply:
Makes sense.. however I think the existing Jenkins job (`leonardo-custom-image`) reads the `Jenkinsfile` in Leonardo `develop` branch at job launch time, so it might break once this PR is merged.

I agree about ideally not having 2 `Jenkinsfiles` checked in though, as we'll not be in this state for very long.

Perhaps a simple way to solve this would be to have the `leonardo-custom-image` job not look at Leo `develop` branch anymore (we could configure it with a different branch/hash).

Reply:
Ah you're right, @rtitle. Thanks for pointing that out! 

I've created `branch-for-legacy-leonardo-custom-image-jenkins-build` off of current `dev`. The [old Jenkins job](https://fc-jenkins.dsp-techops.broadinstitute.org/job/leonardo-custom-image/configure) now points to it. 

This way hopefully we'll be able to just remove that job altogether when ready without having to do anything else.